### ISSUE TITLE
[REGRESSION] Fulltext search partly broken (no result) with master branch

### ISSUE THREAD

**Issue by kelson42** [kiwix/libkiwix#722]
This is what I get with dev master branch:
![image](https://user-images.githubusercontent.com/1029718/157459108-1f63a8e6-a299-45b6-be7e-9382d984b85e.png)

And here the result with release 3.2.0-1:
![image](https://user-images.githubusercontent.com/1029718/157459265-f3efe5d9-41b6-4909-a6b2-df04eba05b28.png)

I suspect one of the last merges to have created a regression

**Comment by kelson42** [comment]
@mgautierfr We should have looked at this one before releasing 10.1.0... but I have forgotten about it :(

**Comment by thavelick** [comment]
I'm experiencing some weird ranking/indexing behavior with kiwix serve vs kiwix desktop that might be related to this.

Pretty much any search I do with `kiwix-serve` ranks things in a way that is pretty much worthless and seemingly random. Whereas `kiwix-desktop` gives me the results I expect. This started happening upon upgrading to 10.1.0.

This screenshot contains kiwix wikipedia searches for `eddie murphy cop`. The results on the left come from `kiwix-serve` and are low quality, the ones on the right come from `kiwix-desktop` and are the high quality I generally am used to:

![20220401_14h03m01s_grim](https://user-images.githubusercontent.com/1319124/161334177-b6807678-9f10-4de3-853b-ef0154694c17.png)

These are from 10.1.0 compiled binaries, but I also got similar results compiling `kiwix-tools` and `libkiwix` from source about a week ago. 

Here are the versions of things I'm on. 

```
~ ➜ kiwix-desktop --version
kiwix-desktop 2.2.1
+ libqt (compile time) 5.15.3
+ libqt (run time) 5.15.3

libkiwix 10.1.0
+ libzim 7.2.0
+ libxapian 1.4.19
+ libcurl 7.82.0
+ libmicrohttpd 0.9.75
+ libz 1.2.11
+ libicu 70.1.0
+ libpugixml 1.12.0

libzim 7.2.0
+ libzstd 1.5.2
+ liblzma 5.2.5
+ libxapian 1.4.18
+ libicu 70.1.0

~ ➜ kiwix-serve --version
kiwix-tools 3.2.0

libkiwix 10.1.0
+ libzim 7.2.0
+ libxapian 1.4.19
+ libcurl 7.82.0
+ libmicrohttpd 0.9.75
+ libz 1.2.11
+ libicu 70.1.0
+ libpugixml 1.12.0

libzim 7.2.0
+ libzstd 1.5.2
+ liblzma 5.2.5
+ libxapian 1.4.18
+ libicu 70.1.0
```

Happy to help on this in any way I can.

**Comment by kelson42** [comment]
@kelson42 Wow... really weird, might be even an other bug!

**Comment by thavelick** [comment]
@kelson42 I'm using `wikipedia_en_all_maxi_2021-12.zim`.

There's a couple things I want to do to further confirm this bug:
1. I want to compile with the latest commits on `master`. It's possible the ones related to search have already fixed this
2. I've used this machine for some personal libkiwix delevelpment. I'm pretty sure I removed all my own stuff and have things installed in common way (via the archlinux aur), but it's possible I've messed up my local environment. Given that I want to spin up a clean VM and ensure this isn't something stupid that I did.

I'll do both tomorrow.

**Comment by thavelick** [comment]
I confirmed the same issue with the docker container for `kiwix-serve` which I believe builds from `master` and nothing weird in my local environment should cause it to act any differntly, so that takes care of both cases in my last comment.

**Comment by thavelick** [comment]
Okay, I've been able to narrow my bug down. This bug does NOT occur with:

```
sudo docker run -v /home/tristan/kiwix:/data -p 8080:80 kiwix/kiwix-serve:3.2.0-1 wikipedia_en_all_maxi_2021-12.zim
```

but it DOES occur with 

```
sudo docker run -v /home/tristan/kiwix:/data -p 8080:80 kiwix/kiwix-serve:3.2.0-2 wikipedia_en_all_maxi_2021-12.zim
```

So judging form the timeline, since it was introduced 5 days ago, my bug is definitely a different bug than the one @kelson42 identifed 24 days ago. Let me know if I should open a separate issue.

**Comment by thavelick** [comment]
Also, if I should open a different ticket, should I open it in the `kiwix-tools` repo? It's a little strange because there hasn't been any commits on `kiwix-tools` in the timeline in question here, so it must be a `libkiwix` issue right?

**Comment by kelson42** [comment]
@thavelick Thx for your investigation. Yes, please open a new ticket, but please make the reproduction steps with released version of Kiwix, so we can be sure this is not something special to your code base.

**Comment by thavelick** [comment]
@kelson42 I opened a dedicated ticket for the issue I found here: https://github.com/kiwix/libkiwix/issues/742

**Comment by mgautierfr** [comment]
This issue seems to be fixed with last published nightly (2022-03-30) and is still broken with the previous one (2022-03-29). This correspond to the merge of #724

**Comment by mgautierfr** [comment]
The root cause is https://github.com/kiwix/libkiwix/pull/620/files#diff-9e3da86d6232708bc11700bf2326552f6ad16b40c029c3ac138470bf4e7d9be7R577-R578 in PR #620 
We do a search for `FT:toto` instead of `toto`. I confirm this is fixed with #724 even if it was not the initial goal.
